src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -158,8 +160,13 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
             // Print a legend message for auto-reference markers used
             if (autoReferenceFound)
             {
-                Console.WriteLine(Strings.ListPkg_AutoReferenceDescription);
+                reportRenderer.WriteLine(Strings.ListPkg_AutoReferenceDescription);
+
+                // Should log into json output ?
             }
+
+            listPackageArgs.Renderer.End();
+            return listPackageArgs.Renderer.ExitCode();
```

Comment:
The renderer having the exit code feels weird. 

Roughly, the design should be something that returns a model and then something else that renders that. 

The model should contain the failure information.